protocol/encoder/encoder.go

```diff
@@ -0,0 +1,261 @@
+package encoder
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/codecrafters-io/kafka-tester/protocol/errors"
+)
+
+type Encoder struct {
+	bytes  []byte
+	offset int
+}
+
+func (re *Encoder) Init(raw []byte) {
+	re.bytes = raw
+	re.offset = 0
+}
+
+func (re *Encoder) PutRawBytesAt(in []byte, offset int, length int) {
```

Comment:
I don't see any of these used at the moment, let's stick to the functions we need at the moment so it's easier to review along w/ usages